### PR TITLE
[1LP][RFR] Embedded Ansible service to check "Cloud Credentials" used in the service.

### PR DESCRIPTION
**Pull request by sbulage**
## Purpose or Intent

- [x] Automated manual tests which has BZ associated with it.

### PRT Run

{{pytest: cfme/tests/ansible/test_embedded_ansible_services.py -k "test_ansible_service_cloud_credentials" --long-running --use-provider=complete -svvvv }}

### CODE REVIEW

File: cfme/tests/ansible/test_embedded_ansible_services.py [hunk 1]
```diff
@@ -854,3 +854,84 @@ def _revert():
...
+    @request.addfinalizer
+    def _revert():
+        with update(ansible_catalog_item):
+            ansible_catalog_item.provisioning = {
+                "playbook": "dump_all_variables.yml",
+                "cloud_type": "<Choose>",
+            }
+
+        service = MyService(appliance, ansible_catalog_item.name)
+        if service_request.exists():
```
Comment (sbulage) [inline cfme/tests/ansible/test_embedded_ansible_services.py]: Comment for reviewers: This tear down is already written in `funcscope` fixture, in this PR #9194

File: cfme/tests/ansible/test_embedded_ansible_services.py [hunk 2]
```diff
@@ -854,3 +854,84 @@ def _revert():
...
+    with update(ansible_catalog_item):
+        ansible_catalog_item.provisioning = {
+            "playbook": playbook,
+            "cloud_type": provider_credentials.credential_type,
+            "cloud_credential": provider_credentials.name,
+        }
+
+    @request.addfinalizer
+    def _revert():
+        with update(ansible_catalog_item):
```
Comment (sshveta) [inline cfme/tests/ansible/test_embedded_ansible_services.py]: Why is this required ? The ansible_catalog_item fixture is deleting the catalog item after yield .
https://github.com/ManageIQ/integration_tests/blob/master/cfme/fixtures/ansible_fixtures.py#L81

File: cfme/tests/ansible/test_embedded_ansible_services.py [hunk 3]
```diff
@@ -854,3 +854,84 @@ def _revert():
...
+                "Provisioning Service "
+                f"[{ansible_catalog_item.name}] from [{ansible_catalog_item.name}]"
+            )
+            service_request_id = appliance.rest_api.collections.service_requests.get(
+                description=request_descr
+            ).id
+            appliance.rest_api.collections.service_requests.action.delete(
+                id=service_request_id
+            )
+        if service.exists:
```
Comment (sshveta) [inline cfme/tests/ansible/test_embedded_ansible_services.py]: Since this is already done as part of fixture in PR 9194 , I suggest using those fixtures as that will make the code a lot smaller

File: cfme/tests/ansible/test_embedded_ansible_services.py (hunk 2 above)
Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_services.py]: `ansible_catalog_item` fixture will be used in different tests. So, resetting back to default values to use it as default fixture for next tests.

File: cfme/tests/ansible/test_embedded_ansible_services.py (hunk 3 above)
Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_services.py]: Yes, I am waiting for the PR #9194 to be merged and after rebasing it will be make code smaller. :smile:

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/tests/ansible/test_embedded_ansible_services.py (hunk 3 above)
Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_services.py]: 9194 is merged.

File: cfme/tests/ansible/test_embedded_ansible_services.py [hunk 4]
```diff
@@ -892,3 +892,59 @@ def _revert():
...
         else "Finished"
     )
+
+
+@pytest.mark.meta(automates=[1444092, 1515561])
+@pytest.mark.parametrize(
+    "credential", CREDENTIALS, ids=[cred[0] for cred in CREDENTIALS]
+)
+@pytest.mark.provider(
+    [RHEVMProvider, EC2Provider, VMwareProvider, AzureProvider], selector=ONE_PER_TYPE
```
[the remark this answers is not in the document]
Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_services.py]: @sshveta I am uncollecting other providers because of Playbooks. Once I will add playbooks for other Providers like OSP, I will Add Marker to collect all Providers.

[the remark this answers is not in the document]
Comment (mshriver) [reply cfme/tests/ansible/test_embedded_ansible_services.py]: Please add in-line comment/TODO for why specific providers are collected.

File: cfme/tests/ansible/test_embedded_ansible_services.py (hunk 2 above)
Comment (mshriver) [reply cfme/tests/ansible/test_embedded_ansible_services.py]: I don't think this is a good practice, as if this fails it will potentially impact all following test cases, and be difficult to debug.  If you are modifying the catalog item, it should be made function scoped, at least for this test case.

Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_services.py]: @mshriver This will impact on Over all tests execution in terms of time. Also, in most of the tests `ansible_catalog_tem` is being modified to use another playbook. Does that makes sense.?

File: cfme/tests/ansible/test_embedded_ansible_services.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_services.py]: I will add TODO/Comment in the #9076 after merge of it.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit e4d0c4a3f80ecc1e0975879098d1df286280bfc3

The *local* fixture **`provider_credentials`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_services.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit a13c08e001250aa8e8447f8622c5c76ec286a6e6

The *local* fixture **`provider_credentials`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_services.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/ansible/test_embedded_ansible_services.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_services.py]: Added Comment/TODO for other providers.

File: cfme/tests/ansible/test_embedded_ansible_services.py (hunk 2 above)
Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_services.py]: @mshriver This PR --> https://github.com/ManageIQ/integration_tests/pull/9490 will take care of this.

Comment (sbulage) [reply cfme/tests/ansible/test_embedded_ansible_services.py]: Also, will push changes to #9490

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 68f7999368da646f60f8efb9afc8003c056037f2

The *local* fixture **`provider_credentials`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_services.py**

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:
